pkg/leanix/integration_hub.go

```diff
@@ -65,27 +65,40 @@ func SelfStartRun(fqdn string, accessToken string, datasource string) (*SelfStar
 	}
 	startResponse := SelfStartResponse{}
 	json.Unmarshal(responseData, &startResponse)
-	_, err = validateConnectorConfiguration(startResponse.ConnectorConfiguration)
-	if err != nil {
-		return nil, err
+	valid := validateConnectorConfiguration(startResponse.ConnectorConfiguration)
+	if valid {
+		log.Info("Connector Configuration is validated")
+		return &startResponse, nil
+	} else {
+		_, err = UpdateFailedProgressStatus(startResponse.ProgressCallbackUrl, "INVALID CONNECTOR CONFIGURATION: ABORTING IHUB RUN.")
+		if err != nil {
+			log.Infof("Failed to update progress[%s] to Integration Hub", FAILED)
+		}
+		return &startResponse, fmt.Errorf("INVALID CONNECTOR CONFIGURATION: CHECK AND RE-RUN WITH VALID CONNECTOR DEFINITION")
```

Comment:
Instead of returning a boolean and running into a if/else construct, you could return something of type error. If it's not nil, then it is valid, otherwise it's not. That way we don't have to check "if valid" but the original "if err != nil"